app/javascript/portal/portalThemeHelper.js

```diff
@@ -0,0 +1,147 @@
+import Cookies from 'js-cookie';
+import { adjustColorForContrast } from '../shared/helpers/colorHelper.js';
+
+const SYSTEM_COOKIE = 'system_theme';
+const SELECTED_COOKIE = 'selected_theme';
+
+export const setPortalHoverColor = theme => {
+  if (window.portalConfig.isPlainLayoutEnabled === 'true') return;
+  const portalColor = window.portalConfig.portalColor;
+  const bgColor = theme === 'dark' ? '#151718' : 'white';
+  const hoverColor = adjustColorForContrast(portalColor, bgColor);
+
+  // Set hover color for border and text dynamically
+  document.documentElement.style.setProperty(
```

Comment:
Do we need to do this in JS? is it not possible in the ERB template?

Reply:
@scmmishra I think we need to handle it from here because we are setting `hoverColor` from color2k.

Reply:
Noted, we can move the implementation to backend anyway, I'm sure there will be a `color2k` like library there. But can be handled later

Reply:
Yeah, I will find out a good gem and use it.